packages/components/src/List/List.tsx

```diff
@@ -2,20 +2,41 @@ import React from "react";
 import classnames from "classnames";
 import get from "lodash/get";
 import groupBy from "lodash/groupBy";
-import styles from "./List.module.css";
+import { Typography } from "@jobber/components/Typography";
+import {
+  BaseListItemProps,
+  ListItem,
+  ListItemProps,
+} from "@jobber/components/List/ListItem";
 import sectionStyles from "./SectionHeader.module.css";
-import { ListItem, ListItemProps } from "./ListItem";
-import { Typography } from "../Typography";
+import styles from "./List.module.css";
 
-interface ListProps {
+interface ListProps<T extends BaseListItemProps = ListItemProps> {
   /**
    * Array of the list items.
    */
-  readonly items: ListItemProps[];
+  readonly items: T[];
+  /**
+   * A function that will be called for each item instead of the default
+   * rendering
+   */
+  readonly customRenderItem?: (item: T) => React.ReactNode;
 }
 
-export function List({ items }: ListProps) {
-  const isSectioned = items.some(item => item.section);
+export function List<T extends BaseListItemProps = ListItemProps>({
+  items,
+  customRenderItem,
+}: ListProps<T>) {
+  if (customRenderItem) {
+    return (
+      <ul className={styles.list}>
+        {items.map(item => (
+          <li key={item.id}>{customRenderItem(item)}</li>
```

Comment:
![image](https://github.com/user-attachments/assets/06831352-8ff6-45e4-8a78-038aef3c12e5)

one thing I'm realizing is that we will need access to the li to pass in all sorts of attributes 😭 

let's imagine we remove the `li` and pass everything to the custom render method... is that ok? we'd have to wrap it in a fragment, and then the consumer has to implement `key` so React doesn't get lost. bleh. that's not ideal.